File: backend/app/schemas/Statement.py
from typing import Optional

from pydantic import BaseModel


class StatementRequest(BaseModel):
    kind: str  # "severance" / "overtime"
    city: str
    monthly_salary: float
    years: Optional[int] = 0  # severance
    months: Optional[int] = 0  # severance
    scenario: Optional[str] = "negotiated"  # severance
    overtime_type: Optional[str] = "weekday"  # overtime
    hours: Optional[float] = 0  # overtime

File: frontend/src/api/statement.js
import http from './http'

// 生成核算单(kind: "severance" 经济补偿 / "overtime" 加班费)
// 参数:城市 + 月薪 + 各类型专属字段(年限/月份/情形 或 加班类型/小时)
export function buildStatement(params) {
  return http.post('/statement', params)
}
